lib/Sema/CSSimplify.cpp

```diff
@@ -4846,6 +4846,30 @@ ConstraintSystem::simplifyKeyPathConstraint(Type keyPathTy,
       return SolutionKind::Solved;
     };
 
+  // We do not allow KeyPaths to go through AnyObject, so let's create a fix
+  // and allow that. This will get diagnosed later.
+  if (auto fixedRootTy =
+          getFixedTypeRecursive(rootTy, subflags, /*wantRValue=*/true)) {
```

Comment:
One last thing in addition to what @slavapestov has already suggested. This whole block has to be moved to around https://github.com/apple/swift/pull/23820/files#diff-69306f7b0822cf52e8c0639307ad7407R4893 since `tryMatchRootAndValueFromKeyPathType` has to be allowed to attempt binding root/leaf with the contextual types first.

Reply:
@xedin Done!

Reply:
Thank you!